q2_moshpit/metabat2/metabat2.py

```diff
@@ -85,30 +90,49 @@ def _process_sample(samp_name, samp_props, common_args, result_loc):
 
         # run metabat2
         bins_dp = _run_metabat2(
-            samp_name, props, tmp, depth_fp, common_args)
+            samp_name, props, tmp, depth_fp, common_args
+        )
 
+        # rename using UUID v4
         all_bins = glob.glob(os.path.join(bins_dp, '*.fa'))
         new_location = os.path.join(str(result_loc), samp_name)
         os.makedirs(new_location)
-        all_bins_new = [_rename_bin(x, new_location) for x in all_bins]
+        all_bins_new = [_rename_bin(new_location) for x in all_bins]
 
         for old, new in zip(all_bins, all_bins_new):
             shutil.move(old, new)
 
 
+def _generate_contig_map(
+        bins: MultiFASTADirectoryFormat
+) -> dict:
+    contig_map = {}
+    for seq in bins.sequences.iter_views(DNAIterator):
+        mag_id = os.path.splitext(os.path.basename(seq[0]))[0]
```

Comment:
The UUID already exists at this point (it's the file name) and I would prefer to keep it this way - otherwise I'd need to do the "moving" here rather than in `_process_sample`, where I think it belongs. Like this this function only needs to care about gathering contig IDs.